### ISSUE TITLE
IOError: [Errno 28] No space left on device

### ISSUE THREAD

**Issue by danbjoseph** [smart-facility/opendronemap-ecs#17]
it's a 797 image dataset

<img width="1131" alt="screen shot 2018-01-02 at 9 00 26 pm" src="https://user-images.githubusercontent.com/4806884/34507006-0c6be4e2-f000-11e7-93c1-705e1416d70a.png">

and

```
[ec2-user@ip-172-31-38-248 ~]$ df
Filesystem     1K-blocks   Used Available Use% Mounted on
/dev/xvda1       8123812 721612   7301952   9% /
devtmpfs        62907328     84  62907244   1% /dev
tmpfs           62915592      0  62915592   0% /dev/shm
```

but

```
2018-01-03 01:54:33,173 Robust matching time : 0.000334024429321s
2018-01-03 01:54:33,173 Full matching 844 / 867, time: 1.12213897705s
2018-01-03 01:54:34,281 20171216_MAVIC101_DJI_0628.JPG - 20171216_SILVER102_DJI_0251.JPG has 1052 candidate matches
2018-01-03 01:54:34,281 Robust matching time : 0.000346899032593s
2018-01-03 01:54:34,281 Full matching 1029 / 1052, time: 1.10797595978s
2018-01-03 01:54:34,391 20171216_MAVIC101_DJI_0743.JPG - 20171216_SILVER102_DJI_0331.JPG has 609 candidate matches
2018-01-03 01:54:34,392 Robust matching time : 0.000371932983398s
2018-01-03 01:54:34,392 Full matching 569 / 609, time: 1.24872303009s
2018-01-03 01:54:35,439 20171216_MAVIC101_DJI_0628.JPG - 20171216_SILVER102_DJI_0188.JPG has 713 candidate matches
2018-01-03 01:54:35,440 Robust matching time : 0.000290870666504s
2018-01-03 01:54:35,440 Full matching 694 / 713, time: 1.15813302994s
2018-01-03 01:54:35,586 20171216_MAVIC101_DJI_0743.JPG - 20171216_MAVIC101_DJI_0799.JPG has 1560 candidate matches
2018-01-03 01:54:35,586 Robust matching time : 0.000333070755005s
2018-01-03 01:54:35,587 Full matching 1547 / 1560, time: 1.19489598274s
[DEBUG]   running PYTHONPATH=/code/SuperBuild/install/lib/python2.7/dist-packages /code/SuperBuild/src/opensfm/bin/opensfm create_tracks /code/opensfm
2018-01-03 01:54:36,275 reading features
2018-01-03 01:54:56,257 Merging features onto tracks
2018-01-03 01:55:33,399 Good tracks: 596153
Traceback (most recent call last):
  File "/code/SuperBuild/src/opensfm/bin/opensfm", line 34, in <module>
    command.run(args)
  File "/code/SuperBuild/src/opensfm/opensfm/commands/create_tracks.py", line 31, in run
    data.save_tracks_graph(tracks_graph)
  File "/code/SuperBuild/src/opensfm/opensfm/dataset.py", line 352, in save_tracks_graph
    save_tracks_graph(fout, graph)
  File "/code/SuperBuild/src/opensfm/opensfm/dataset.py", line 528, in save_tracks_graph
    str(image), str(track), fid, x, y, r, g, b))
IOError: [Errno 28] No space left on device
Traceback (most recent call last):
  File "/code/run.py", line 47, in <module>
    plasm.execute(niter=1)
  File "/code/scripts/opensfm.py", line 115, in process
    (context.pyopencv_path, context.opensfm_path, tree.opensfm))
  File "/code/opendm/system.py", line 34, in run
    raise Exception("Child returned {}".format(retcode))
Exception: Child returned 1
ERROR: Parameter problem: Nothing to upload.
ERROR: Parameter problem: Nothing to upload.
```

**Comment by matthewberryman** [comment]
df isn't that helpful here as the container would have been cleaned up by that time. Definitely running out of space—the problem is that unless the extra EBS volume is mounted and ECS is configured to use it (which isn't in the way I've set up the task), then it's the root volume size that needs to be big enough. I've updated the docs in https://github.com/OpenDroneMap/opendronemap-ecs/commit/20d0a521ed68b1c54391258ac81097def163ff69 - you will need to terminate your existing autoscaling cluster (wait for it to finish termination so there's no name conflict on launching new config with same autoscaling group name), then create a copy of your ODM launch config and adjust the volumes to only have a Root volume with enough space.
